uplc.md

```diff
@@ -0,0 +1,205 @@
+# Untyped Plutus Core
+
+Plutus smart contracts are written in Haskell, and are ultimately[^il] compiled to a language known as **Untyped Plutus Core** (UPLC). Unlike Haskell, UPLC is a low-level and untyped language. It supports only a handful of builtin types and functions, that are then strung together in lambda applications (see [lambda calculus](https://en.wikipedia.org/wiki/Lambda_calculus)).
+
+Here is a simple example of a UPLC program that is equivalent to the Haskell `id` function:
+
+```lisp
+Program
+  ()
+  (Version () 1 0 0)
+  (LamAbs
+     ()
+     (DeBruijn {dbnIndex = 0})
+     (Var () (DeBruijn {dbnIndex = 1})))
+```
+
+## Pluto
+
+As you might have noticed it is not practical to write UPLC by hand. [Pluto](https://github.com/Plutonomicon/pluto) is a simple programming language that assembles directly to UPLC; that is to say, it 'maps' directly (more or less) to the AST of UPLC. Understanding Pluto facilitates an understanding of UPLC. Here's the equivalent Pluto program for the above UPLC example:
+
+```haskell
+(\× -> x)
```

Comment:
weird `x` here

Reply:
Looks to be the Agda vscode extension getting in the way ...